astropy/modeling/astro_sherpa.py

```diff
@@ -0,0 +1,433 @@
+
+    '''
+    So the generall method is to dstack(and transpose each row) so
+        [x1,x2,x3],[y1,y2,y3],[z1,z2,z3],...
+    becomes
+        [x1,y1,,z1],[x2,y2,,z2],[x3,y3,,z3],...
+        the and put it into a data set! this assumes that all columns are set for all datasets!
+        athough a none can be put in place of missing columns
+    '''
+    if z is None:
+        if xerr is None:
+            if yerr is None:
+                _data = Data1D("wrapped_data", x=x, y=y)
+            else:
+                _data = Data1D("wrapped_data", x=x, y=y, staterror=yerr)
+        else:
+            if yerr is None:
+                _data = Data1DInt("wrapped_data", xlo=x - xerr, xhi=x + xerr, y=y)
+            else:
+                _data = Data1DInt("wrapped_data", xlo=x - xerr, xhi=x + xerr, y=y, staterror=yerr)
+    else:
+        if xerr is None and yerr is None:
+            if zerr is None:
+                _data = Data2D("wrapped_data", x0=x, x1=y, y=z)
+            else:
+                _data = Data2D("wrapped_data", x0=x, x1=y, y=z, staterror=zerr)
+        elif xerr is not None and yerr is not None:
+            if zerr is None:
+                _data = Data2DInt("wrapped_data", x0lo=x - xerr, x0hi=x + xerr, x1lo=y - yerr, x1hi=y + yerr, y=z)
+            else:
+                _data = Data2DInt("wrapped_data", x0lo=x - xerr, x0hi=x + xerr, x1lo=y - yerr, x1hi=y + yerr, y=z, staterror=zerr)
+        else:
+            raise Exception("Set xerr and yerr, or set neither!")
+
+    return _data
+
+
+def make_datasets(n_dim, x, y, z=None, xerr=None, yerr=None, zerr=None):
+    '''
+    Parameters
+        ----------
+        n_dim: int
+            Used to veirfy required number of dimentions.
+        x : array (or list of arrays)
+            input coordinates
+        y : array (or list of arrays)
+            input coordinates
+        z : array (or list of arrays) (optional)
+            input coordinates
+        xerr : array (or list of arrays) (optional)
+            an array of errors in x
+        yerr : array (or list of arrays) (optional)
+            an array of errors in y
+        zerr : array (or list of arrays) (optional)
+            an array of errors in z
+
+    returns:
+        _data: a sherpa dataset
+    '''
+
+    x = np.array(x)
+    y = np.array(y)
+
+    if x.ndim == 2 or (x.dtype == np.object or y.dtype == np.object):
+        data = []
+        if z is None:
+            z = len(x) * [None]
+
+        if xerr is None:
+            xerr = len(x) * [None]
+
+        if yerr is None:
+            yerr = len(y) * [None]
+
+        if zerr is None:
+            zerr = len(z) * [None]
+
+        for nn, (xx, yy, zz, xxe, yye, zze) in enumerate(zip(x, y, z, xerr, yerr, zerr)):
+            data.append(_make_dataset(n_dim, x=xx, y=yy, z=zz, xerr=xxe, yerr=yye, zerr=zze, n=nn))
+        return DataSimulFit("wrapped_data", data)
```

Comment:
As above, return an instance of a wrapper class (a different one than the one returned by _make_datasets) for `DataSimulFit`, rather than `DataSimulFit` itself.